cpp/example_code/new_s3/source/list_objects.cpp

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * snippet-sourcedescription:[list_objects.cpp demonstrates how to list the objects in an Amazon S3 bucket.]
+ * snippet-keyword:[C++]
+ * snippet-sourcesyntax:[cpp]
+ * snippet-keyword:[Code Sample]
+ * snippet-keyword:[Amazon S3]
+ * snippet-service:[s3]
+ * snippet-sourcetype:[full-example]
+ * snippet-sourcedate:[]
+ * snippet-sourceauthor:[AWS]
+ */
+
+#include <awsdoc/s3/s3_examples.h>
+#include <aws/core/Aws.h>
+#include <aws/core/utils/memory/stl/AWSString.h>
+#include <aws/s3/S3Client.h>
+#include <aws/s3/model/ListObjectsRequest.h>
+
+bool AwsDoc::S3::ListObjects(const char* bucketName)
+{
+    std::cout << "Objects in S3 bucket: " << bucketName << std::endl;
+
+    Aws::S3::S3Client s3Client;
+
+    Aws::S3::Model::ListObjectsRequest listObjectsRequest;
+    listObjectsRequest.WithBucket(bucketName);
+
+    auto listObjectsOutcome = s3Client.ListObjects(listObjectsRequest);
+
+    if (listObjectsOutcome.IsSuccess())
+    {
+        Aws::Vector<Aws::S3::Model::Object> objectList =
+            listObjectsOutcome.GetResult().GetContents();
+
+        for (auto const &s3Object : objectList)
+        {
+            std::cout << "* " << s3Object.GetKey() << std::endl;
+        }
+        return true;
+    }
+    else
+    {
+        std::cout << "ListObjects error: " <<
+            listObjectsOutcome.GetError().GetExceptionName() << " " <<
+            listObjectsOutcome.GetError().GetMessage() << std::endl;
+        return false;
+    }
+}
+
+int main(int argc, char** argv)
+{
+    Aws::SDKOptions options;
+    Aws::InitAPI(options);
+    {
+        if (argc < 2)
```

Comment:
More a matter of style, but any concerns about shortening "int main(int argc, char** argv)" to simply "int main()" and dropping usage of argc and argv? I think it would be more compact and readable by hard-coding some example values to help in understanding and to cut down on the command-line parsing.

Reply:
I think it's fine as long as you don't need to provide parameters to the main function.